Add subscriptions to relationship events to docs (#2467)
neo4j/graphql: docs/modules/ROOT/pages/subscriptions/index.adoc

* **xref::subscriptions/events/create.adoc[Create]** - Listen for newly created nodes.
* **xref::subscriptions/events/update.adoc[Update]** - Listen for changes to existing nodes.
* **xref::subscriptions/events/delete.adoc[Delete]** - Listen for deleted nodes.
* **xref::subscriptions/events/create_relationship.adoc[Create Relationship]** - Listen for newly created relationships.
* **xref::subscriptions/events/delete_relationship.adoc[Delete Relationship]** - Listen for deleted relationships.

> All events will be triggered individually, with one event per node created, updated or deleted, or relationship created or deleted, even in events caused by the same mutation.

Reviewer: Quite a hard sentence to phrase, but I think this sounds a little cleaner!

```suggestion
All events will be triggered individually, even in events caused by the same Mutation. Events are fired on the creation or deletion of nodes and relationships, and on the update of nodes.
```
